src/charm.py

```diff
@@ -0,0 +1,363 @@
+            self,
+            jobs=[
+                {
+                    "static_configs": [{"targets": [f"*:{PROMETHEUS_PORT}"]}],
+                }
+            ],
+        )
+
+    def _on_install(self, event: InstallEvent) -> None:
+        """Handles the install event.
+
+        Args:
+            event (InstallEvent): Juju event.
+        """
+        if not self._container.can_connect():
+            self.unit.status = WaitingStatus("Waiting for container to be ready")
+            event.defer()
+            return
+        if not self._storage_is_attached():
+            self.unit.status = WaitingStatus("Waiting for storage to be attached")
+            event.defer()
+            return
+        self._write_ue_config_file()
+
+    def _configure_sdcore_smf(self, event: EventBase) -> None:
+        """Adds pebble layer and manages Juju unit status.
+
+        Args:
+            event: Juju event
+        """
+        for relation in ["default-database", "smf-database", "fiveg_nrf"]:
+            if not self._relation_created(relation):
+                self.unit.status = BlockedStatus(
+                    f"Waiting for `{relation}` relation to be created"
+                )
+                return
+        if not self._container.can_connect():
+            self.unit.status = WaitingStatus("Waiting for container to be ready")
+            return
+        if not self._default_database_is_available():
+            self.unit.status = WaitingStatus(
+                "Waiting for `default-database` relation to be available"
+            )
+            return
+        if not self._smf_database_is_available():
+            self.unit.status = WaitingStatus("Waiting for `smf-database` relation to be available")
+            return
+        if not self._nrf_is_available():
+            self.unit.status = WaitingStatus("Waiting for NRF relation to be available")
+            return
+        if not self._storage_is_attached():
+            self.unit.status = WaitingStatus("Waiting for storage to be attached")
+            event.defer()
+            return
+        if not self._ue_config_file_is_written():
+            event.defer()
+            self.unit.status = WaitingStatus(
+                f"Waiting for `{UEROUTING_CONFIG_FILE}` config file to be pushed to workload container"  # noqa: W505, E501
+            )
+            return
+        self._update_config_file()
+        self._container.add_layer("smf", self._pebble_layer, combine=True)
+        self._container.replan()
+        self.unit.status = ActiveStatus()
+
+    def _update_config_file(self) -> None:
+        """Updates config file.
+
+        Writes the config file if it does not exist or
+        if the content does not match.
+        """
+        content = self._render_config_file(
+            default_database_name=DEFAULT_DATABASE_NAME,
+            default_database_url=self._smf_database_data()["uris"].split(",")[0],
```

Comment:
`_smf_database_data`? Shouldn't this be fetching from the `default_database`?

Reply:
Fixed. Really good catch! The utests didn't catch it because it is partially patched. 
I deployed the whole thing, relate and exec into it and It was setting a proper Mongo URL, I don't know how querying Mongo works, but It would still work, the database is the same, the only change is the collection, and the endpoint should be the same. I didn't spend much time, if you have any guess I'm curious. Thanks again!